rdtools/filtering.py

```diff
@@ -124,3 +140,322 @@ def csi_filter(poa_global_measured, poa_global_clearsky, threshold=0.15):
+        a pandas datetime index.
+    mounting_type : String
+        String representing the mounting configuration associated with the
+        AC power/energy time series. Can either be "Fixed" or "Tracking".
+        Default set to 'Fixed'.
+    Returns
+    -------
+    pd.Series
+        AC power time series with renamed 'datetime' column, and
+        renamed 'value' column.
+    """
+    # Check that it's a Pandas series with a datetime index.
+    # If not, raise an error.
+    if not isinstance(power_ac.index, pd.DatetimeIndex):
+        raise TypeError('Must be a Pandas series with a datetime index.')
+    # Check the other input variables to ensure that they are the
+    # correct format
+    if (mounting_type != "Tracking") & (mounting_type != "Fixed"):
+        raise ValueError(
+            "Variable mounting_type must be string 'Tracking' or 'Fixed'.")
+    # Get the names of the series and the datetime index
+    column_name = power_ac.name
+    if column_name is None:
+        column_name = 'value'
+        power_ac = power_ac.rename(column_name)
+    index_name = power_ac.index.name
+    if index_name is None:
+        index_name = 'datetime'
+        power_ac = power_ac.rename_axis(index_name)
+    return power_ac
+
+
+def calculate_max_derivative(power_ac, roll_periods):
+    """
+    This function calculates the maximum derivative over a rolling
+    time period for an AC power time series. A pandas series of
+    the rolling derivative is returned.
+    Parameters
+    ----------
+    power_ac : pd.Series
+        Pandas time series, representing PV system power or energy with
+        a pandas datetime index.
+    roll_periods: Int
+        Number of readings to calculate the rolling maximum derivative on.
+    Returns
+    -------
+    pd.Series
+        Time series of the rolling maximum derivative.
+    """
+    # Calculate the maximum value over a forward-rolling window
+    max_roll = power_ac.iloc[::-1].rolling(roll_periods).max()
+    max_roll = max_roll.reindex(power_ac.index)
+    # Calculate the minimum value over a forward-rolling window
+    min_roll = power_ac.iloc[::-1].rolling(roll_periods).min()
+    min_roll = min_roll.reindex(power_ac.index)
+    # Calculate the maximum derivative within the foward-rolling window
+    derivative_max = (max_roll - min_roll) / ((max_roll + min_roll) / 2) * 100
+    return derivative_max
+
+
+def logic_clip_filter(power_ac,
+                      mounting_type='Fixed',
+                      derivative_cutoff=0.2,
+                      roll_periods=None):
+    '''
+    This filter is a logic-based filter that is used to filter out
+    clipping periods in AC power and AC energy time series.
+    The AC power or energy time series is filtered based on the
+    maximum derivative over a rolling window, as compared to a user-set
+    derivate_cutoff (default set to 0.2).  The size of the
+    rolling window is increased when the system is a tracked system.
+    Parameters
```

Comment:
```suggestion
    rolling window is increased when the system is a tracked system.

    Parameters
```

Sphinx is very particular about whitespace in docstrings -- without this blank line, the first parameter doesn't render correctly: https://rdtools--200.org.readthedocs.build/en/200/generated/rdtools.filtering.logic_clip_filter.html